src/shared/util.js

```diff
@@ -371,6 +371,20 @@ export function getCardsMissingCount(deck, grpid) {
   return mainMissing + sideboardMissing;
 }
 
+export function getMissingCardCounts(deck) {
+  const missingCards = {};
+  const allCardIds = new Set(
+    [...deck.mainDeck, ...deck.sideboard].map(card => card.id)
+  );
+  allCardIds.forEach(grpid => {
+    const missing = getCardsMissingCount(deck, grpid);
+    if (missing) {
+      missingCards[grpid] = missing;
+    }
```

Comment:
```suggestion
missingCards[grpid] = getCardsMissingCount(deck, grpid);
```

`getCardsMissingCount` always returns a number, so `if(missing)` always passes.

Reply:
True, but I actually intended this one in the arithmetic sense. Updating the logic for clarity.

Reply:
I see the check for `> 0` in the new version of the file - actually, you don't need that. Either `getMissingCardCount === 0`, in which case it is perfectly fine to set that to `missingCards[grpid]` (it is the default value anyway), or you want to set the value you just computed :)

Reply:
The goal here is to produce a sparsely populated object to make looping over missing cards more efficient. We only want to add non-zero missing cards to the result.

Reply:
Didn't see that `missingCards` is a dictionary and not an object, you're right of course!